vnext/Desktop/WebSocketModule.cpp

```diff
@@ -98,7 +103,7 @@ void WebSocketModule::SendEvent(string&& eventName, dynamic&& args)
     instance->callJSFunction("RCTDeviceEventEmitter", "emit", dynamic::array(std::move(eventName), std::move(args)));
 }
 
-IWebSocket* WebSocketModule::GetWebSocket(int64_t id, string&& url)
+IWebSocket* WebSocketModule::GetOrCreateWebSocket(int64_t id, string&& url)
```

Comment:
>IWebSocket* [](start = 0, length = 11)

Since this is never null, it should return by reference
IWebSocket&

Reply:
The underlying storage is an `std::vector<std::unique_ptr<IWebSocekt>>`.
Wouldn't de-referencing be more expensive performance-wise than just passing the underlying raw-pointer?

Reply:
No, internally, reference is passed as just a pointer (void*), so perf is the same. Difference being the expectation of the user code. The correct pattern is that when explicit pointers are passed around, the consuming code is required to check for nullptr before doing anything. Apis which use references imply that user code can just use objects without additional checks.

---
In reply to: [298413042](https://github.com/microsoft/react-native-windows/pull/2662#discussion_r298413042) [](ancestors = 298413042)